lib/metadata/jsonMetadata.hpp

```diff
@@ -0,0 +1,151 @@
+const std::string RFI_NUM_BAD_INPUTS("RFI_NUM_BAD_INPUTS"); // a uint32_t of bad frames count
+const std::string RFI_FLAGGED_SAMPLES("RFI_FLAGGED_SAMPLES"); // a int32_t of FPGA frames flagged as containing RFI
+const std::string LOST_TIMESAMPLES("LOST_TIMESAMPLES"); // a int32_t of samples lost
+const std::string STREAM_ID("STREAM_ID");               // a uint64_t stream identifier set originally by the FPGA board
+
+const std::string FIRST_PACKET_RECV_TIME("FIRST_PACKET_RECV_TIME"); // The system time when the first packet in the frame was captured
+const std::string TV_SEC("TV_SEC");                     // the tv_sec memmber of a timeval
+const std::string TV_USEC("TV_USEC");                   // the tv_usec memmber of a timeval
+
+
+struct beamCoord {
+    float right_ascension[MAX_NUM_BEAMS];
+    float declination[MAX_NUM_BEAMS];
+    uint32_t scaling[MAX_NUM_BEAMS];
+
+    beamCoord() {
+#ifdef DEBUG
+        for(auto& ra : right_ascension)
+            ra = std::nanf("");
+        for(auto& dec : declination)
+            dec = std::nanf("");
+        for(auto& scale : scaling)
+            scale = 0;
+#endif
+    }
+
+    // TODO: turn into a from_json function?
+    beamCoord(const metadata& md) {
+        {
+        const auto& ra(md[RIGHT_ASCENSION]);
+        if(ra.size() > MAX_NUM_BEAMS)
+                throw std::runtime_error("Number of beams request exceeds MAX_NUM_BEAMS");
+        size_t i = 0;
+        for(auto it = ra.cbegin() ; it != ra.cend() ; ++it) {
+            it->get_to(right_ascension[i++]);
+        }
+#ifdef DEBUG
+        for(size_t i = ra.size() ; i < MAX_NUM_BEAMS ; ++i) {
+            right_ascension[i] = std::nanf("");
+        }
+#endif
+        }
+
+        {
+        const auto& dec(md[DECLINATION]);
+        if(dec.size() > MAX_NUM_BEAMS)
+                throw std::runtime_error("Number of beams request exceeds MAX_NUM_BEAMS");
+        size_t i = 0;
+        for(auto it = dec.cbegin() ; it != dec.cend() ; ++it) {
+                it->get_to(declination[i]);
+        }
+#ifdef DEBUG
+        for(size_t i = dec.size() ; i < MAX_NUM_BEAMS ; ++i) {
+            declination[i] = std::nanf("");
+        }
+#endif
+        }
+
+        {
+        const auto& scale(md[SCALING]);
+        if(scale.size() > MAX_NUM_BEAMS)
+            throw std::runtime_error("Number of beams request exceeds MAX_NUM_BEAMS");
+        size_t i = 0;
+        for(auto it = scale.cbegin() ; it != scale.cend() ; ++it) {
+            it->get_to(scaling[i]);
+        }
+#ifdef DEBUG
+        for(size_t i = scale.size() ; i < MAX_NUM_BEAMS ; ++i) {
+            scaling[i] = 0;
+        }
+#endif
+        }
+    }
+};
+
+struct coarseFreq {
+        int nfreq;
+        int coarse_freq[CHORD_META_MAX_FREQ];
+
+        // TODO: turn into a from_json function?
+        coarseFreq(const metadata& md) {
+        md[NFREQ].get_to(nfreq);
+        const auto& freq(md[COARSE_FREQ]);
+        if(freq.size() > MAX_NUM_BEAMS)
```

Comment:
Shouldn't this be max frequencies not beams.   But also I think we should be able to get rid of hard coded max sizes since we are backing the metadata with a dynamic structure.